mne/channels/channels.py

```diff
@@ -805,11 +833,17 @@ def interpolate_bads(
         else:
             eeg_mne = True
         _interpolate_bads_meeg(
-            self, mode=mode, origin=origin, eeg=eeg_mne, exclude=exclude
+            self, mode=mode, origin=origin, eeg=eeg_mne, exclude=exclude, method=method
         )
-        _interpolate_bads_nirs(self, exclude=exclude)
+        _interpolate_bads_nirs(self, exclude=exclude, method=method)
 
         if reset_bads is True:
+            if "nan" in method.values():
+                warn(
+                    "interpolate_bads was called with method='nan' and "
+                    "reset_bads=True. Information about interpolated channels"
+                    "will not exist. Consider setting reset_bads=False"
```

Comment:
```suggestion
                    "reset_bads=True. Consider setting reset_bads=False so that the "
                    "nan-containing channels can be easily excluded from later "
                    "computations."
```